### PR TITLE
convert object type to string

### PR DESCRIPTION
**Pull request by yunjie-hub**

### CODE REVIEW

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 1]
```diff
@@ -167,6 +167,15 @@ def _extract_data_and_correlation_id(df: DataFrame, extract_correlation_id: bool
     otherwise, return the dataref content which is a url to the json file.
     """
 
+    def convert_object_to_str(dataframe):
```
Comment (RichardLi1437) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:170]: I suppose this is a pandas dataframe? maybe add type

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 2]
```diff
@@ -167,6 +167,15 @@ def _extract_data_and_correlation_id(df: DataFrame, extract_correlation_id: bool
...
+        columns = dataframe.columns
+        index = 0
+        for t in dataframe.dtypes:
+            if t == "object":
+                dataframe[columns[index]] = dataframe[columns[index]].astype(str)
+                index += 1
```
Comment (feng-pan8014) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:176]: should this be outside of "if"?

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 3]
```diff
@@ -167,6 +167,15 @@ def _extract_data_and_correlation_id(df: DataFrame, extract_correlation_id: bool
     otherwise, return the dataref content which is a url to the json file.
     """
 
+    def convert_object_to_str(dataframe):
+        columns = dataframe.columns
+        index = 0
+        for t in dataframe.dtypes:
```
Comment (ycheng35xo) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:173]: nit: can use this to get the index in the for loop

for index, t in enumerate(dataframe.dtypes):

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py (hunk 2 above)
Comment (yunjie-hub) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py]: Change to column name

**Review by feng-pan8014 — DISMISSED** [review]

**Review by RichardLi1437 — APPROVED** [review]

**Pull request merged**